Added TryToContinue exception handling to cmsRun

Dr15Jones opened the pull request:
#### PR description:

Removed SkipEvent and FailPath exception handling in favor of TryToContinue. The new handling option will cause a job to continue on event processing if an exception associated to TryToContinue is thrown. For the event which had the exception, 
- If a module is marked as `shouldTryToContinue()` in the configuration, it will be eligible to be run even if a TryToContinue exception happens. Marked modules will also not propagate a TryToContinue exception to the rest of the system whether the exception happened during prefetching or while running the module. If a marked module  is on a Path, it will not propagate the exception to the Path. However, if another module on the Path is dependent on the throwing module and is not marked then the Path will still fail (see next item).
- all Paths with a module that is directly or indirectly dependent on the module which threw will be marked as having an exception. Any modules on the Path which are directly or indirectly dependent upon the throwing module will not be run, unless the dependent module is marked `shouldTryToContinue()`. Any modules on the Path not already scheduled will not be run, including those marked as `shouldTryToContinue()`. Those which were scheduled and independent of the thrown module will still be run.
- Unscheduled modules which depend on the throwing module will not be run, unless they are marked as `shouldTryToContinue()`. Unscheduled modules which are independent may still be run if their data is requested.

#### PR validation:

All the framework unit tests (including the newly added ones) pass.

Dr15Jones committed:
Minor improvements to TestGlobalAnalyzers

- move exception from destructor to endJob
- allow silencing of cout calls.

Dr15Jones committed:
Added TryToContinue exception handling option

- removed FailPath and SkipEvent options

Dr15Jones committed:
Added shouldTryToContinue configuration option

This works with the TryToContinue exception option. A module with shouldTryToContinue turned on will
- not propagate any exception in the TryToContinue list (i.e. will appear to run normally)
- will be called in the case a dependent module throws an exception in the TryToContinue list.

Dr15Jones committed:
Set Path to exception state when using TryToContinue

The exception is used to set the state of the Path but is not propagated further.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42441/36449

- This PR adds an extra 136KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/ParameterSet/python/Config.py  modified in PR(s): #37996

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42441/36449/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42441/36449/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

Dr15Jones committed:
Code formating

Dr15Jones commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42441/36450

- This PR adds an extra 136KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/ParameterSet/python/Config.py  modified in PR(s): #37996

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones (Chris Jones) for master.

It involves the following packages:

- FWCore/Framework (**core**)
- FWCore/Integration (**core**)
- FWCore/ParameterSet (**core**)


@makortel, @smuzaffar, @Dr15Jones can you please review it and eventually sign? Thanks.
@makortel, @missirol, @wddgit this is something you requested to watch as well.
@perrotta, @dpiparo, @rappoccio you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78df49/34025/summary.html
**COMMIT**: 096bb5a779f6da33e92333683ed6320bb4f76e4d
**CMSSW**: CMSSW_13_3_X_2023-08-01-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/42441/34025/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found errors in the following unit tests:

<pre>---> test testSSTGainPCL_fromRECO had ERRORS
---> test testDiMuonVertexMonitor had ERRORS
---> test DRNTest had ERRORS
---> test checkMultiRunHarvestingOutput had ERRORS
and more ...
</pre>

## Comparison Summary



**Summary**:
 - You potentially removed 29 lines from the logs
 - Reco comparison results: 16 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3150821
 - DQMHistoTests: Total failures: 9
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3150790
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-08-01-1100+78df49/58206/triggerResults): no differences found

Dr15Jones committed:
Updated tests to not use SkipEvent or FailPath

Dr15Jones commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42441/36455

- This PR adds an extra 20KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/ParameterSet/python/Config.py  modified in PR(s): #37996

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78df49/34038/summary.html
**COMMIT**: 9a7195794909aa461c62e068672b076185e86ef5
**CMSSW**: CMSSW_13_3_X_2023-08-02-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/42441/34038/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially added 9 lines to the logs
 - Reco comparison results: 8 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3150821
 - DQMHistoTests: Total failures: 110
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3150689
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-08-02-1100+78df49/58216/triggerResults): no differences found

tvami commented:
hi @Dr15Jones could you please squash the commits? I also see a comment "//TODO have to deal with 'cms.catch'" is that for a follow-up PR?

Dr15Jones commented:
@tvami I'd rather not squash the commits as each has a purpose [except for the code formatting].

As for the `//TODO` for `cms.catch` those changed were already applied and I just failed to notice them before doing the commit. <sigh/>

perrotta commented:
> As for the `//TODO` for `cms.catch` those changed were already applied and I just failed to notice them before doing the commit.

Then could you please remove them, so that future maintainers will not get confused by them?

Dr15Jones commented:
> Then could you please remove them, so that future maintainers will not get confused by them?

I will. I'll be doing myself a favor since I am most likely the 'future maintainer' :).

emanueleusai commented:
are the DQM differences in tracking spurious? the code does not seem to touch any tracking modules beyond alca, but I wonder if they come from events that were previously being skipped now just raising TryToContinue

Dr15Jones commented:
> are the DQM differences in tracking spurious? the code does not seem to touch any tracking modules beyond alca, but I wonder if they come from events that were previously being skipped now just raising TryToContinue

I double checked and all the RelVals just use the default setting for the exception handling which is to always rethrow any exception seen (i.e. FailPath, SkipEvent and IgnoreCompletely are always empty). Therefore this code change could not be the cause of any spurious change in the DQM.

makortel reviewed (commented):
Could you add tests also for these cases
- a `TryToContinue` exception is thrown from a non-Event function
- a `shouldTryToContinue()` module throws an `TryToContinue` exception from a non-Event function
- an ESModule throws a `TryToContinue` exception
  - when the job does not have any `shouldTryToContinue()` EDModule
  - when the job has a `shouldTryToContinue()` EDModule that does not consume the ESProduct of the ESModule
  - when the job has a `shouldTryToContinue()` EDModule that consumes the ESProduct of the ESModule
- a Source throws a `TryToContinue` exception (say during Event processing)
- an OutputModule throws a `TryToContinue` exception (say during Event processing, in both cases of `shouldTryToContinue()` and without)
- check that the `TryToContinue` exception message is printed out

Is the current testing sufficient for ensuring the Path with a throwing module is marked in "error" state in `TriggerResults`? (in principle the OutputModule `SelectEvents` uses that, but I'm wondering if a more explicit check would be useful)

According to the plan https://github.com/cms-sw/cmssw/issues/41512#issuecomment-1615284641 all those should result in job termination.

Would the test you did in https://github.com/cms-sw/cmssw/issues/41512#issuecomment-1664127254 be worthwhile to include as unit test?

Dr15Jones commented:
We are getting another round of comments before making further changes. 

As a reminder to myself, we also determined `TryToContinue` can cause problems if ROOT fast copying is used. (This problem was also missed for the old `SkipEvent` cases). We will want to disable fast copying if `TryToContinue` is in use.

Dr15Jones committed:
Code improvements based on review feedback

- cleaned up comments
- unit test now functions and tests non shouldTryToContinue case

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42441/36480

- This PR adds an extra 52KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQMOffline/Alignment/test/DiMuonTkAlDQMHarvester_cfg.py  modified in PR(s): #42461
  - File  DQMOffline/Alignment/test/DiMuonTkAlDQMValidator_cfg.py  modified in PR(s): #42461
  - File  FWCore/Framework/interface/StreamSchedule.h  modified in PR(s): #42461
  - File  FWCore/Framework/interface/maker/Worker.h  modified in PR(s): #42461
  - File  FWCore/Framework/src/StreamSchedule.cc  modified in PR(s): #42461
  - File  FWCore/ParameterSet/python/Config.py  modified in PR(s): #37996

cmsbuild commented:
Pull request #42441 was updated. @perrotta, @smuzaffar, @consuegs, @Dr15Jones, @micsucmed, @nothingface0, @emanueleusai, @clacaputo, @tvami, @cmsbuild, @makortel, @saumyaphor4252, @pmandrik, @syuvivida, @tjavaid, @mandrenguyen, @francescobrivio, @rvenditti can you please check and sign again.

Dr15Jones commented:
hold

we have opened a conversation in issue about possible alternative user interfaces for setting the `shouldTryToContinue`.

cmsbuild commented:
Pull request has been put on hold by  @Dr15Jones
They need to issue an `unhold` command to remove the `hold` state or L1 can `unhold` it for all

Dr15Jones committed:
Switch to modulesToCallForTryToContinue in options

The specification is in Process.options instead of a method for
all Module types.

Dr15Jones commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42441/36507

- This PR adds an extra 68KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQMOffline/Alignment/test/DiMuonTkAlDQMHarvester_cfg.py  modified in PR(s): #42461
  - File  DQMOffline/Alignment/test/DiMuonTkAlDQMValidator_cfg.py  modified in PR(s): #42461
  - File  FWCore/Framework/interface/StreamSchedule.h  modified in PR(s): #42461
  - File  FWCore/Framework/interface/maker/Worker.h  modified in PR(s): #42461
  - File  FWCore/Framework/src/StreamSchedule.cc  modified in PR(s): #42461
  - File  FWCore/ParameterSet/python/Config.py  modified in PR(s): #37996

cmsbuild commented:
Pull request #42441 was updated. @perrotta, @smuzaffar, @consuegs, @Dr15Jones, @micsucmed, @nothingface0, @emanueleusai, @clacaputo, @tvami, @makortel, @saumyaphor4252, @pmandrik, @syuvivida, @tjavaid, @mandrenguyen, @francescobrivio, @rvenditti can you please check and sign again.

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78df49/34143/summary.html
**COMMIT**: 5fc7187e0ab7541e4cca5e7f899a1113831db002
**CMSSW**: CMSSW_13_3_X_2023-08-07-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/42441/34143/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found errors in the following unit tests:

<pre>---> test TestFWCoreParameterSetDriver had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially removed 3 lines from the logs
 - Reco comparison results: 6 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3150947
 - DQMHistoTests: Total failures: 3
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3150922
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-08-07-1100+78df49/58296/triggerResults): no differences found

Dr15Jones commented:
please test

cmsbuild commented:
Pull request #42441 was updated. @perrotta, @smuzaffar, @consuegs, @Dr15Jones, @micsucmed, @nothingface0, @emanueleusai, @clacaputo, @tvami, @makortel, @saumyaphor4252, @pmandrik, @syuvivida, @tjavaid, @mandrenguyen, @francescobrivio, @rvenditti can you please check and sign again.

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78df49/34163/summary.html
**COMMIT**: 106c65189feb14311f6b4216a416de94e7dc587d
**CMSSW**: CMSSW_13_3_X_2023-08-08-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/42441/34163/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found errors in the following unit tests:

<pre>---> test testConnectionPool had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially removed 4 lines from the logs
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3150947
 - DQMHistoTests: Total failures: 3
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3150922
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-08-08-1100+78df49/58320/triggerResults): no differences found

emanueleusai commented:
please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78df49/34219/summary.html
**COMMIT**: 106c65189feb14311f6b4216a416de94e7dc587d
**CMSSW**: CMSSW_13_3_X_2023-08-10-2300/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/42441/34219/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found errors in the following unit tests:

<pre>---> test TestGeneratorInterfacePythia8ConcurrentGeneratorFilter had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially removed 24 lines from the logs
 - Reco comparison results: 10 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3150947
 - DQMHistoTests: Total failures: 6
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3150919
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-08-10-2300+78df49/58352/triggerResults): no differences found

Dr15Jones commented:
The same test failure is seen in the IB

emanueleusai commented:
+1

- unrelated uint test failure

Dr15Jones commented:
unhold

tvami commented:
@cmsbuild , please test
 * the unit test should be fixed in the IBs now

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78df49/34329/summary.html
**COMMIT**: 106c65189feb14311f6b4216a416de94e7dc587d
**CMSSW**: CMSSW_13_3_X_2023-08-16-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/42441/34329/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially removed 9 lines from the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3152915
 - DQMHistoTests: Total failures: 0
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3152893
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-08-16-1100+78df49/58473/triggerResults): no differences found

Dr15Jones committed:
Additional unit tests for TryToContinue

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42441/36632

- This PR adds an extra 208KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/ParameterSet/python/Config.py  modified in PR(s): #37996

cmsbuild commented:
Pull request #42441 was updated. @perrotta, @smuzaffar, @consuegs, @Dr15Jones, @tvami, @makortel, @nothingface0, @emanueleusai, @clacaputo, @mandrenguyen, @cmsbuild, @saumyaphor4252, @pmandrik, @syuvivida, @tjavaid, @micsucmed, @francescobrivio, @rvenditti can you please check and sign again.

makortel commented:
@cmsbuild, please test

makortel commented on FWCore/Integration/test/BuildFile.xml:
```diff
@@ -113,10 +113,21 @@
+  <test name="TestFWCoreIntegrationShouldTryToContinueRun" command="cmsRun ${LOCALTOP}/src/FWCore/Integration/test/test_shouldTryToContinue_cfg.py -- --inRun 2>&amp;1 | fgrep 'exception for testing purposes'"/>
+  <test name="TestFWCoreIntegrationShouldTryToContinueLumi" command="cmsRun ${LOCALTOP}/src/FWCore/Integration/test/test_shouldTryToContinue_cfg.py -- --inLumi 2>&amp;1 | fgrep 'exception for testing purposes'"/>
   <test name="TestFWCoreIntegrationShouldTryToContinueIndirect" command="cmsRun ${LOCALTOP}/src/FWCore/Integration/test/test_shouldTryToContinue_cfg.py -- --indirect"/>
+  <test name="TestFWCoreIntegrationShouldTryToContinueIndirect" command="cmsRun ${LOCALTOP}/src/FWCore/Integration/test/test_shouldTryToContinue_cfg.py -- --indirect"/>
```
Is this test now being run twice?

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78df49/34344/summary.html
**COMMIT**: de2c4c77877cd4ecdcba93090c9a6025cbb21aef
**CMSSW**: CMSSW_13_3_X_2023-08-17-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/42441/34344/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found errors in the following unit tests:

<pre>---> test testSiStripHitResolution had ERRORS
---> test testCalibTrackerSiStripCommon had ERRORS
---> test testSiStripHitEfficiency had ERRORS
---> test testAlignmentStats had ERRORS
and more ...
</pre>

## Comparison Summary



**Summary**:
 - You potentially removed 2 lines from the logs
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3152915
 - DQMHistoTests: Total failures: 0
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3152893
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-08-17-1100+78df49/58485/triggerResults): no differences found

cmsbuild commented:
Pull request #42441 was updated. @perrotta, @smuzaffar, @consuegs, @Dr15Jones, @tvami, @makortel, @nothingface0, @emanueleusai, @clacaputo, @mandrenguyen, @cmsbuild, @saumyaphor4252, @pmandrik, @syuvivida, @tjavaid, @micsucmed, @francescobrivio, @rvenditti can you please check and sign again.

makortel commented:
@cmsbuild, please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78df49/34354/summary.html
**COMMIT**: 27df62ddfa8b0d999a83be4f560d4ea8dc0812dd
**CMSSW**: CMSSW_13_3_X_2023-08-18-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/42441/34354/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found errors in the following unit tests:

<pre>---> test testSiStripHitResolution had ERRORS
---> test testSiStripHitEfficiency had ERRORS
---> test testCalibTrackerSiStripCommon had ERRORS
---> test testAlignmentStats had ERRORS
and more ...
</pre>

## Comparison Summary



**Summary**:
 - You potentially removed 8 lines from the logs
 - Reco comparison results: 6 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3152915
 - DQMHistoTests: Total failures: 3
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3152890
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-08-18-1100+78df49/58499/triggerResults): no differences found

makortel commented:
These unit tests are already failing in IB

makortel commented:
+core

tvami commented:
+alca

Dr15Jones commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78df49/34384/summary.html
**COMMIT**: 27df62ddfa8b0d999a83be4f560d4ea8dc0812dd
**CMSSW**: CMSSW_13_3_X_2023-08-21-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/42441/34384/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially removed 9 lines from the logs
 - Reco comparison results: 6 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3152915
 - DQMHistoTests: Total failures: 3
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3152890
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-08-21-1100+78df49/58519/triggerResults): no differences found

mandrenguyen commented:
+1

emanueleusai commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @antoniovilela, @rappoccio (and backports should be raised in the release meeting by the corresponding L2)

rappoccio commented:
+1
